metronome/checkpointing/interpreter/src/io/iohk/metronome/checkpointing/interpreter/messages/InterpreterMessage.scala

```diff
@@ -167,9 +180,135 @@ object InterpreterMessage extends RPCMessageCompanion {
       with FromService
       with NoResponse
 
+  /** Interpreter <=> Chain */
+
+  /** The interpreter requests the chain for block at target height in order to construct
+    * the service block
+    */
+  case class CheckpointBlockRequest(
+      requestId: RequestId,
+      blockNumber: BigInt
+  ) extends InterpreterMessage
+      with Request
+      with FromInterpreter
+
+  /** The chain replies to @CheckpointBlockRequest with block metadata */
+  case class CheckpointBlockResponse(
+      requestId: RequestId,
```

Comment:
I think there is no need to model the interaction between the ETC chain and the interpreter with `RPCMessage` machinery, i.e. the `requestId`. That's because this is for _Remote_ Procedure Calls and the interpreter, according to the current design, will run inside the ETC process, in other words it's a _local_ call, so it can simply be a `Task` for example, modeled as a trait, similar to the `InterpreterRPC` and `ServiceRPC` interfaces.